R/geom-function.R

```diff
@@ -45,8 +45,11 @@ geom_function <- function(mapping = NULL, data = NULL, stat = "function",
                           position = "identity", ..., na.rm = FALSE,
                           show.legend = NA, inherit.aes = TRUE) {
   # Warn if supplied data is going to be overwritten
-  if (!is.null(data) && identical(stat, "function")) {
-    warn("`data` is not used by stat_function()")
+  if (identical(stat, "function")) {
```

Comment:
You can in theory provide a ggproto object directly... This check will fail if a user for some reason pass in `StatFunction` (I know this is an esoteric issue)

Reply:
Yes, I think it's fine, since the worst case scenario is no warning.